contracts/AelinPool.sol

```diff
@@ -385,6 +445,83 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+
+    function purchasePoolTokensWithNft(
+        address _collectionAddress,
+        uint256[] calldata _tokenIds,
+        uint256 _purchaseTokenAmount
+    ) external lock {
+        require(hasNftList, "pool does not have an NFT list");
+        require(block.timestamp < purchaseExpiry, "not in purchase window");
+        require(!nftWalletUsedForPurchase[msg.sender], "wallet already used");
```

Comment:
this should only be a requirement when the pool is one investment amount per wallet. Otherwise, the same wallet can invest multiple times by passing in token ids

Reply:
5e0426fb0e68557fda5f49de0412911a8bbed85b